graylog2-server/src/main/java/org/graylog/plugins/views/search/validation/fields/UnknownFieldsIdentifier.java

```diff
@@ -17,20 +17,24 @@
 package org.graylog.plugins.views.search.validation.fields;
 
 import org.graylog.plugins.views.search.rest.MappedFieldTypeDTO;
+import org.graylog.plugins.views.search.validation.ParsedQuery;
 import org.graylog.plugins.views.search.validation.ParsedTerm;
+import org.graylog.plugins.views.search.validation.QueryValidator;
+import org.graylog.plugins.views.search.validation.ValidationMessage;
 import org.graylog.plugins.views.search.validation.ValidationRequest;
+import org.graylog.plugins.views.search.validation.ValidationType;
 import org.graylog2.indexer.fieldtypes.MappedFieldTypesService;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 @Singleton
-public class UnknownFieldsIdentifier {
+public class UnknownFieldsIdentifier implements QueryValidator {
```

Comment:
I'd consider keeping original UnknownFieldsIdentifier and introducing separate UnknownFieldsValidator, which just uses the identifier. The current approach combines two responsibilities in this class (parsing and validating) and forces you to to use parsing in the unit test, making this unit test not a pure "unit test"...

Reply:
The query parsing is now moved outside of the scope of this class. The identification of unknown fields is needed only to for the validation, so I see it as one responsibility only. And the scope is so small that I'd rather keep everything inside one class than to split it further into two different. I have refactored the code since your comment. Is it for better of worse from your POV?

Reply:
Ok!